lib/galaxy/datatypes/isa.py

```
@@ -63,57 +63,39 @@ def utf8_text_file_open(path):
 logger.addHandler(ch)
 logger.setLevel(logging.ERROR)
 
-
 # ISA class {{{1
 ################################################################
 
 class Isa(data.Data):
     """ Base class for implementing ISA datatypes """
-    file_ext = "isa"
     composite_type = 'auto_primary_file'
     allow_datatype_change = False
     is_binary = True
+    _main_file_regex = None
 
-    # Add static metadata responsible for naming dataset instances
-#    metadata.MetadataElement(name="base_name", desc="Generic dataset name", default="isa dataset",
-#                             readonly=True, set_in_upload=True)
-
-    @classmethod
-    def _make_investigation(cls, filename):
-        
-        # Parse JSON file
-        if filename[-5:].lower() == '.json':
-            fp = utf8_text_file_open(filename)
-            isa = isajson.load(fp)
-            return isa
-        
-        # Parse ISA-Tab investigation file
-        parser = isatab.InvestigationParser()
-        fp = utf8_text_file_open(filename)
-        parser.parse(fp)
-        isa = parser.isa
-        return isa
+    # Make investigation instance {{{2
+    ################################################################
+    
+    def _make_investigation_instance(self, filename):
+        raise NotImplementedError()
 
     # Constructor {{{2
     ################################################################
 
-    def __init__(self, **kwd):
-        data.Data.__init__(self, **kwd)
+    def __init__(self, main_file_regex, **kwd):
+        super(Isa, self).__init__(**kwd)
+        self._main_file_regex = main_file_regex
```

We should `re.compile` this argument here to fail early in case of bugs.